Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -97,21 +116,43 @@ export class ArtifactEngine {
 
             if (tl.match([pathToMatch], this.patternList, null, matchOptions).length > 0) {
                 Logger.logInfo("Processing " + item.path);
-                sourceProvider.getArtifactItem(item).then((contentStream) => {
-                    Logger.logInfo("Got download stream for item: " + item.path);
-                    destProvider.putArtifactItem(item, contentStream)
-                        .then((item) => {
-                            this.artifactItemStore.updateState(item, models.TicketState.Processed);
+                var downloadedFromCache = false;
```

Comment:
@omeshp we will miss downloading the metadata file if the pattern doesn't include it. How should we handle this case?

Reply:
ggod point, lets add it to createPatternList incase incremental download is enabled